packages/core/upload/admin/src/components/EditAssetDialog/index.js

```diff
@@ -52,7 +52,12 @@ export const EditAssetDialog = ({
 
   const handleSubmit = async values => {
     if (asset.isLocal) {
-      const nextAsset = { ...asset, ...values };
+      const nextAsset = {
+        ...asset,
+        ...values,
+        // update name of File object
+        rawFile: new File([asset.rawFile], values.name, { type: asset.rawFile.type }),
+      };
```

Comment:
Hi @iicdii,

Thanks for your contribution, your changes do solve the file name problem. However, the point is that the other information in the edit form is not sent. Also, the values from the form are already in the object and this makes duplicates. Some changes at a lower level of the file upload are needed to solve this without changing the data. 

I'm closing this PR for the moment because the problem is not completely solved here.